### PR TITLE
[BugFix] fix decimalV2 array_slice crash bug

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #
Multi parameters decimalV2 function will match decimalV3 function, but doesn't set decimalV3 precision&scale.
user input may: array_slice(decimalV2, tinyint)
sr always match array_slice(decimalV3, bigint)

because array_slice(decimalV2, tinyint) can't match array_slice(decimalV2, bigint) in IS_IDENTICAL mode, but
only match array_slice(decimalV3, bigint) in IS_NONSTRICT_SUPERTYPE_OF mode (decimalV3 function is higher
than decimalV2 function)

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java|13|14|92.86%|[300]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|2|2|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/DecimalTypeTest.java [hunk 1]
```diff
@@ -212,4 +211,27 @@ public void testArrayAggDecimal() throws Exception {
             connectContext.getSessionVariable().setNewPlanerAggStage(stage);
         }
     }
+
+    @Test
+    public void testDecimalV2ArraySlice() throws Exception {
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/DecimalTypeTest.java:216]: Why not use SQL-test to covert this case?

Comment (Seaven) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/DecimalTypeTest.java:216]: already have case

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java [hunk 2]
```diff
@@ -287,6 +286,41 @@ public static boolean argumentTypeContainDecimalV3(String fnName, Type[] argumen
                 .anyMatch(t -> t.getItemType().isDecimalV3());
     }
 
+    // Multi parameters decimalV2 function will match decimalV3 function, and doesn't set decimalV3 precision&scale.
+    // like that, user input may: array_slice(decimalV2, tinyint), sr always match array_slice(decimalV3, bigint),
+    // because array_slice(decimalV2, tinyint) can't match array_slice(decimalV2, bigint) in IS_IDENTICAL mode, but
+    // only match array_slice(decimalV3, bigint) in IS_NONSTRICT_SUPERTYPE_OF mode (decimalV3 function is higher
+    // than decimalV2 function).
+    public static boolean argumentTypeContainDecimalV2(String fnName, Type[] argumentTypes) {
+        if (!FunctionSet.ARRAY_SLICE.equals(fnName)) {
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java:295]: Dose any other any other array function have similar problem?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java [hunk 3]
```diff
@@ -287,6 +286,41 @@ public static boolean argumentTypeContainDecimalV3(String fnName, Type[] argumen
...
+            return false;
+        }
+
+        if (Arrays.stream(argumentTypes).anyMatch(Type::isDecimalV2)) {
+            return true;
+        }
+
+        // check array child type
+        return Arrays.stream(argumentTypes).filter(Type::isArrayType).map(t -> (ArrayType) t)
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java:304]: what if the nested array type?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java (hunk 2 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java:295]: only array_slice

**Review by packy92 — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java (hunk 3 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java:304]: nest array type will use polymorphic function, it's right

**Review by liuyehcf — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**